packages/data-explorer/src/main.ts

```diff
@@ -74,18 +74,32 @@ Vue.config.productionTip = false
 // This way the mainView can coordinate the events
 Vue.prototype.$eventBus = new Vue()
 
+// @ts-ignore
+const { lng, fallbackLng } = window.__INITIAL_STATE__
+
+const router = new Router({
+  base: process.env.NODE_ENV === 'production' ? packageJson.name : process.env.BASE_URL,
+  routes
+})
+
+// Navigation guard that makes sure that all explorer
+// query params are present.
+router.beforeEach((to, from, next) => {
+  globalThis.defaultRouteQuery = defaultRouteQuery
+  globalThis.to = to.query
+  if (to.name === 'de-view') {
+    const missingQueryParams = Object.keys(defaultRouteQuery).filter((i) => !Object.keys(to.query).includes(i))
+    if (missingQueryParams.length) {
+      next({ path: `/explorer/${to.params.entity}`, query: defaultRouteQuery })
+    } else { next() }
+  } else { next() }
+})
+
 Vue.use(i18n, {
-  lng: 'en',
-  fallbackLng: 'en',
-  namespace: ['dataexplorer'],
+  lng,
+  fallbackLng,
```

Comment:
What happens if window.__INITIAL_STATE__ is empty. Maybe we need a in application fallback 'en'?

Reply:
Good one! Restored the original situation where lng and fallbackLng were set in main.ts